docs/extensions.rst

```diff
@@ -0,0 +1,10 @@
+.. automodule:: corehq.extensions
+
+Extension points
```

Comment:
I tried to use autodocs to add docs for the extension points but it didn't work well (presumably because the functions are replaced by the callables).